networking/cluster_network_overview.py

```diff
@@ -0,0 +1,312 @@
+import boto3
+from kubernetes import client, config
+from collections import defaultdict
+from tabulate import tabulate
+import os, sys
+from datetime import datetime
+
+
+def create_usage_bar(current, total, bar_length=10):
+    try:
+        percentage = (current / total) * 100
+        filled_length = int(bar_length * current // total)
+        bar = '█' * filled_length + '-' * (bar_length - filled_length)
+        return f'|{bar}| {percentage:.1f}%'
+    except Exception as e:
+        print(f"Error creating usage bar: {str(e)}")
+        return "N/A"
+
+def get_subnet_usage_info(ec2_client, subnet_id):
+    print(f"Fetching subnet usage info for subnet {subnet_id}...")
+    try:
+        subnet_info = ec2_client.describe_subnets(SubnetIds=[subnet_id])['Subnets'][0]
+        availability_zone = subnet_info['AvailabilityZone']
+        cidr_block = subnet_info['CidrBlock']
+        total_ips = 2 ** (32 - int(subnet_info['CidrBlock'].split('/')[1])) - 5
+        available_ips = subnet_info['AvailableIpAddressCount']
+        used_ips = total_ips - available_ips
+        percentage_used = (used_ips / total_ips) * 100
+        percentage_free = 100 - percentage_used
+
+        return [
+            availability_zone,
+            cidr_block,
+            total_ips,
+            used_ips,
+            available_ips,
+            create_usage_bar(used_ips, total_ips)
+        ]
+    except Exception as e:
+        print(f"Error fetching subnet usage info for subnet {subnet_id}: {str(e)}")
+        return ["N/A"] * 6
+
+def get_cluster_subnet_data(eks_client, ec2_client, cluster_name):
+    print(f"Fetching subnet data for cluster {cluster_name}...")
+    try:
+        cluster_info = eks_client.describe_cluster(name=cluster_name)
+        cluster_subnet_ids = cluster_info['cluster']['resourcesVpcConfig']['subnetIds']
+
+        cluster_subnet_info = defaultdict(list)
+        for subnet_id in cluster_subnet_ids:
+            cluster_subnet_info[subnet_id] = get_subnet_usage_info(ec2_client, subnet_id)
+
+        return cluster_subnet_info
+    except Exception as e:
+        print(f"Error fetching cluster subnet data for cluster {cluster_name}: {str(e)}")
+        return defaultdict(list)
+
+
+def get_cluster_worker_node_ips(v1, ec2_client):
+    print("Fetching EC2 IPs for the EKS cluster...")
+    worker_node_ips = set()
+    subnet_ids = set()
+    worker_subnet_info = defaultdict(list)
+
+    try:
```

Comment:
added loop to fetch all pods for using " _continue" and a "continue_token" since the call "list_node" is paginated. Using "metadata._continue" to check if there is more data to retrieve.